src/test/kotlin/com/expediagroup/sdk/core/config/ConfigurationDefinitionTest.kt

```diff
@@ -453,6 +493,50 @@ class ConfigurationDefinitionTest {
         val throwable = assertThrows<ConfigurationException> {
             configurationDefinition.parse(props)
         }
-        assertEquals(throwable.message, "value is not uppercase")
+        assertEquals("value is not uppercase", throwable.message)
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = [true, false])
+    fun `maps correct value when configuration value is boolean`(value: Boolean) {
+        val configurationDefinition = ConfigurationDefinition()
+        val booleanConfigurationName = "boolean_configuration"
+        val configurationDocumentation = "This is a test configuration documentation"
+        val configurationImportance = ConfigurationKey.Importance.HIGH
+        configurationDefinition.define(
+            name = booleanConfigurationName,
+            documentation = configurationDocumentation,
+            type = ConfigurationKey.Type.BOOLEAN,
+            importance = configurationImportance
+        )
+
+        assertDoesNotThrow {
+            val mappedConfigurations = configurationDefinition.parse(mapOf(booleanConfigurationName to value))
+            assertNotNull(mappedConfigurations)
+            assertTrue(mappedConfigurations[booleanConfigurationName] is Boolean)
+            assertEquals(value, mappedConfigurations[booleanConfigurationName])
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = ["true", "false", "TRUE", "FALSE", "TruE", "FalsE"])
```

Comment:
Do we really need the parameters after he first 2?

Reply:
Yes, because old code used to be case-insensitive. So after refactoring and adding more tests for code coverage, I had to make sure that the new code is still also code-insensitive and that no changes to the logic occurred.